### PR TITLE
Pass X-Api-Key for GRPC streaming client

### PR DESCRIPTION
**Pull request by denisbsu**
https://proxima-one.atlassian.net/browse/PRX-1331

### CODE REVIEW

File: src/streamdb/consumerClient.ts [hunk 1]
```diff
@@ -13,12 +13,26 @@ import { PausableStream, StreamController } from "../lib/pausableStream";
...
+    const isSecure = uri.includes(":443");
+    const channelCredentials = isSecure
       ? grpc.credentials.createSsl()
       : grpc.credentials.createInsecure();
-    this.consumer = new StreamConsumerServiceClient(uri, credentials, {
+    const callCredentials = grpc.credentials.createFromMetadataGenerator(
+      (opts, callback) => {
+        const metadata = new grpc.Metadata();
+        if (apiKey !== undefined) {
+          metadata.add("x-api-key", apiKey ?? "");
```
Comment (kalabukdima) [inline src/streamdb/consumerClient.ts:25]: Unnecessary `?? ""`

File: src/client/proximaStreamClient.ts [hunk 2]
```diff
@@ -133,11 +135,15 @@ export class ProximaStreamClient {
   private getStreamConsumerClient(endpoint: string) {
     return (
       this.clients[endpoint] ??
-      (this.clients[endpoint] = new StreamDBConsumerClient(endpoint))
+      (this.clients[endpoint] = new StreamDBConsumerClient(
```
Comment (kalabukdima) [inline src/client/proximaStreamClient.ts:138]: Is this the only place where the client is created internally? I thought that the registry client can also create a consumer client sometimes :thinking:

File: src/client/proximaStreamClient.ts [hunk 3]
```diff
@@ -12,13 +12,15 @@ export class ProximaStreamClient {
...
   private readonly offsetsCache: NodeCache;
+  private readonly apiKey?: string;
 
   public constructor(
     private readonly options: StreamClientOptions = {
       registry: new StreamRegistryClient(),
     }
   ) {
     this.registry = options.registry ?? new StreamRegistryClient();
+    this.apiKey = options.apiKey;
```
Comment (kalabukdima) [inline src/client/proximaStreamClient.ts:23]: If someone passed in a registry with api-key set, we should probably extract it from that registry.

**Review by kalabukdima — CHANGES_REQUESTED** [review]

File: src/client/proximaStreamClient.ts (hunk 2 above)
Comment (denisbsu) [reply src/client/proximaStreamClient.ts:138]: only one as far as I can see

File: src/client/proximaStreamClient.ts (hunk 3 above)
Comment (denisbsu) [reply src/client/proximaStreamClient.ts]: updated

**Review by kalabukdima — APPROVED** [review]

**Pull request merged**